### DISCUSSION TITLE
Concern about a few deprecated functions for V4.X (mostly trim)

### DISCUSSION THREAD

**Discussion by sma27** [jquery/jquery#5529] [General]
A huge fan of jQuery and thank you for working on an update to the library. In reading about the version 4 release, I have some concerns about the methods that will be removed, especially trim().

There are certain functions that were so simple and so handy that most developer's code is littered with them. Functions like trim are one of them. Although most browsers support String.trim(), it does not work the same way as jQuery did. You can't trim a number, object, null, or undefined value. This causes several problems for migration:

1. It isn't just a find and replace. Removing a few lines of code from jQuery will create a huge burden on developers to make code modifications and test for breakages. 
2. Many developers use jQuery plugins that may or may not be actively developed/supported. Most of these plugins use jQuery functions like trim, isArray, isNumber, etc. Why not just wrap the JS native functions and provide the same behavior that jQuery has traditionally supported? Developers can use the slimmed down version if they choose.

I don't buy into the hype that jQuery is no longer relevant, or that it's footprint is too big. Some of my projects would require 2X the amount of JS code and 2X the time to pull off the same functionality. I love jQuery and plan to keep using it!

**Comment by sma27** [comment]
Assuming these functions will be removed. This is the code that will end up being ported from 3.7 into my projects to avoid breaking plugins and legacy code.

```javascript

// Option A: Used as is from jQuery 3.7
if (typeof jQuery.trim !== "function") {
    const rtrim = /^[\s\uFEFF\xA0]+|([^\s\uFEFF\xA0])[\s\uFEFF\xA0]+$/g;
    
    jQuery.trim = function (text) {
        return text == null ?
            "" :
            (text + "").replace(rtrim, "$1");
    }
}
``` 

```javascript

// Option B: Wrap native String.trim()
if (typeof jQuery.trim !== "function") {
    jQuery.trim = function (text) {
        return text == null ?
            "" :
            (text + "").trim();
    }
}
```

**Comment by timmywil** [comment]
Thank you for opening this! Similar points were made in [2019](https://github.com/jquery/jquery/issues/4363) and when it was [ultimately removed](https://github.com/jquery/jquery/pull/4695). However,

1. It's outside the scope of jQuery and the non-string behavior was never documented.
1. The migrate plugin will help migrate user code and give a [good option for replacement](https://github.com/jquery/jquery-migrate/blob/main/warnings.md#trim-jqmigrate-jquerytrim-is-deprecated-use-stringprototypetrim).
1. If it's common enough to make migration a pain, a user plugin can be added instead, much like your comment suggested. This has the advantage of keeping unused code out of jQuery for all those that migrate to the native option.
